presto-main/src/main/java/com/facebook/presto/cost/PlanNodeCostEstimate.java

```diff
@@ -42,38 +57,95 @@ public static PlanNodeCostEstimate networkCost(double networkCost)
         return new PlanNodeCostEstimate(0, 0, networkCost);
     }
 
+    private final double individualCpuCost;
+    private final double individualMemoryCost;
+    private final double individualNetworkCost;
     private final double cpuCost;
     private final double memoryCost;
     private final double networkCost;
 
-    public PlanNodeCostEstimate(double cpuCost, double memoryCost, double networkCost)
+    public PlanNodeCostEstimate(double individualCpuCost, double individualMemoryCost, double individualNetworkCost)
```

Comment:
this constructor may be useful for testing, but in general we should use the one below.

Reply:
I left this constructor here because it is used in several other places and seems to be used to pass in the initial costs. If this constructor were to disappear, we would be explicitly passing in the same costs to both the individual and cumulative parameters. Is that what you're getting at?

Reply:
oh i see it becomes cummulative after the "add" method.  I would put a comment making that clearer and also make the other constructor private.  Additionally instead of having two constructors that initialize stuff, call the other constructor from here.